src/tinuous/travis.py

```diff
@@ -114,13 +114,27 @@ def get_commit(self, build: Dict[str, Any], event_type: EventType) -> Optional[s
             return commit
         elif event_type is EventType.PULL_REQUEST:
             pr = self.ghrepo.get_pull(build["pull_request_number"])
-            if pr.merge_commit_sha == build["commit"]["sha"]:
+            if pr.merged:
+                # Once a PR is merged, the merge_commit_sha changes to the
+                # actual merge commit rather than the autogenerated merge
+                # commit used by Travis.  We thus check whether the build
+                # commit (the autogenerated merge commit) has the PR head as a
+                # parent; if so, we assume that the build commit was the
+                # autogenerated merge commit for this PR head and return the PR
+                # head.
+                build_commit = self.ghrepo.get_commit(build["commit"]["sha"])
+                if any(c.sha == pr.head.sha for c in build_commit.parents):
+                    return pr.head.sha
```

Comment:
this would work for builds which correspond to the latest `pr.head` , but wouldn't work for prior builds, right?
Couldn't we assume that the 2nd among parents is the one which was merged into the target branch and thus would be the one of the PR head at that point in time?

Reply:
I know of no reason we can assume that.

Reply:
well, we know that
- it is travis
- it is a PR and it was merged
- we have a merge commit (2 parents)
  - one of them must be the head commit which was merged

so the only question is only on the order of commits in parents:  I think it should be safe to assume that for a merged [*] PR with 2 parents, 2nd one is the desired commit since I bet it is just the listing as known to git commit and there 2nd one would be the one merged into the 1st one.

[*] PRs could be "merged" with other strategies:
- `squash&merge`
- `rebase&merge`

in both of those cases we cannot assume that 2nd commit would be the head. Do we have information about what type of merge it was from github or travis?  if we do -- we can IMHO safely proceed to assume the 2nd commit to be a head sha for regular merges, and indeed fall through to `UNK` for 2 other types

Reply:
Assumption incorporated into code.  For the record, because we're looking up autogenerated merge commits rather than final merge commits, they should always be the non-squash, non-rebase kind and thus always have two parent commits (and if one doesn't, `{commit}` will be set to "UNK").